source/polygon.cpp

```diff
@@ -0,0 +1,46 @@
+ld perimeter(vector<Point<ll>>& poli){
+    ld res = 0.0;
+    int n = ssize(poli);
+    for(int i = 1; i <= n; ++i) res += poli[(i-1) % n].dist(poli[(i)%n]);
```

Comment:
`(i-1) % n` can be simply `i-1`, since `i <= n` the modulo will never be needed. Also, no need for parentheses in `(i)%n`

Reply:
fixed